components/camel/camel-jms/src/main/java/org/switchyard/component/camel/jms/deploy/CamelJmsInboundHandler.java

```diff
@@ -0,0 +1,68 @@
+package org.switchyard.component.camel.jms.deploy;
+
+import java.util.HashMap;
+
+import javax.annotation.Resource;
+import javax.xml.namespace.QName;
+import javax.xml.ws.WebServiceContext;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.ValueBuilder;
+import org.apache.camel.impl.ExpressionAdapter;
+import org.apache.camel.model.RouteDefinition;
+import org.switchyard.ServiceDomain;
+import org.switchyard.common.camel.SwitchYardCamelContext;
+import org.switchyard.common.type.Classes;
+import org.switchyard.component.camel.common.CamelConstants;
+import org.switchyard.component.camel.common.handler.InboundHandler;
+import org.switchyard.component.camel.common.handler.MessageComposerProcessor;
+import org.switchyard.component.camel.common.handler.OperationSelectorProcessor;
+import org.switchyard.component.camel.jms.model.CamelJmsBindingModel;
+import org.switchyard.runtime.event.ExchangeCompletionEvent;
+
+
+public class CamelJmsInboundHandler extends InboundHandler<CamelJmsBindingModel> {
+
+    @Resource
+    private WebServiceContext _wsContext;
```

Comment:
camel-jms binding shouldn't depend on WebServiceContext. The security credentials should be extracted in service binding, and should be carried as a binding agnostic format when propagating.